shared-module/time/__init__.c

```diff
@@ -42,4 +45,11 @@ uint64_t common_hal_time_monotonic_ns(void) {
 
 void common_hal_time_delay_ms(uint32_t delay) {
     mp_hal_delay_ms(delay);
+    // if the delay was cut short by a CTRL-C then clear the keyboard exception
+    if(MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)))
```

Comment:
Instead of putting this in `common_hal_time_delay_ms()`, to me it makes sense just to hoist it back into `mp_hal_delay_ms()`, which already does pending exception checking:
https://github.com/adafruit/circuitpython/blob/beb80e75458168dd475f2153179bd786c867f380/supervisor/shared/tick.c#L102-L109